pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -54,6 +60,24 @@ func (cd *clusterDeploy) sync(key string, cluster *v3.Cluster) error {
 	original := cluster
 	cluster = original.DeepCopy()
 
+	// create cluster role for this cluster's templates and templateversions
+	// check if this role already exists
+	crObtained, err := cd.clusterroles.Get("cluster-"+cluster.Name+"-use-templates-templateversions", metav1.GetOptions{})
+
+	newCR := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster-" + cluster.Name + "-use-templates-templateversions",
+		},
+	}
+
+	// create cluster role only if it doesn't already exist
+	if !reflect.DeepEqual(crObtained, newCR) {
```

Comment:
same as for project level catalog. Only create when not found in cache, no need for deep equal